### ISSUE TITLE
[Question] How do i know if my multiple account are 1) logged in and 2) being used?

### ISSUE THREAD

**Issue by wardude202** [AHAAAAAAA/PokemonGo-Map#3141]
Question in title

Im using 3 accounts
2x PTC accounts
and a google account

how can i tell if im using all 3?

**Comment by Syndiction** [comment]
I'm trying to utilize the same type of login order, and i can't tell either. I"m on the dev branch and i couldn't find an exact tutorial on the setup

**Comment by blindreaper** [comment]
Also does the bot login into accounts that have disconnected?

**Comment by Silvikk** [comment]
My command to develop  - python runserver.py -a ptc -u user1 -u user2 -u user3 -p pass1 -p pass2 -p pass3 -l "lat, long" -st 9
and here we go

2016-08-02 15:31:41,239 [ search_worker_1][        search][    INFO] Searching s
tep 36, remaining 181
2016-08-02 15:31:41,627 [ search_worker_3][        models][    INFO] Upserted 0
pokemon, 54 pokestops, and 5 gyms
2016-08-02 15:31:41,957 [ search_worker_2][        search][    INFO] Searching s
tep 37, remaining 180
2016-08-02 15:31:42,319 [ search_worker_1][        models][    INFO] Upserted 1
pokemon, 54 pokestops, and 5 gyms
2016-08-02 15:31:43,068 [ search_worker_2][        models][    INFO] Upserted 0
pokemon, 52 pokestops, and 5 gyms
2016-08-02 15:31:43,069 [ search_worker_0][        search][    INFO] Searching s
tep 38, remaining 179
2016-08-02 15:31:43,868 [ search_worker_0][        models][    INFO] Upserted 0
pokemon, 52 pokestops, and 5 gyms
2016-08-02 15:31:44,437 [ search_worker_4][        search][    INFO] Searching s
tep 39, remaining 178
2016-08-02 15:31:45,246 [ search_worker_4][        models][    INFO] Upserted 0
pokemon, 56 pokestops, and 5 gyms
2016-08-02 15:31:46,839 [ search_worker_3][        search][    INFO] Searching s
tep 40, remaining 177
2016-08-02 15:31:47,421 [ search_worker_1][        search][    INFO] Searching s
tep 41, remaining 176
2016-08-02 15:31:47,539 [ search_worker_3][        models][    INFO] Upserted 0
pokemon, 25 pokestops, and 3 gyms
2016-08-02 15:31:48,170 [ search_worker_1][        models][    INFO] Upserted 0
pokemon, 25 pokestops, and 3 gyms
2016-08-02 15:31:48,191 [ search_worker_2][        search][    INFO] Searching s
tep 42, remaining 175
2016-08-02 15:31:49,069 [ search_worker_2][        models][    INFO] Upserted 0
pokemon, 25 pokestops, and 3 gyms
2016-08-02 15:31:49,135 [ search_worker_0][        search][    INFO] Searching s
tep 43, remaining 174
2016-08-02 15:31:50,005 [ search_worker_0][        models][    INFO] Upserted 0
pokemon, 56 pokestops, and 5 gyms
2016-08-02 15:31:50,362 [ search_worker_4][        search][    INFO] Searching s
tep 44, remaining 173
2016-08-02 15:31:51,177 [ search_worker_4][        models][    INFO] Upserted 0
pokemon, 56 pokestops, and 5 gyms
2016-08-02 15:31:52,667 [ search_worker_3][        search][    INFO] Searching s
tep 45, remaining 172
2016-08-02 15:31:53,361 [ search_worker_1][        search][    INFO] Searching s
tep 46, remaining 171
2016-08-02 15:31:53,542 [ search_worker_3][        models][    INFO] Upserted 0
pokemon, 30 pokestops, and 4 gyms
2016-08-02 15:31:54,236 [ search_worker_1][        models][    INFO] Upserted 0
pokemon, 30 pokestops, and 4 gyms
2016-08-02 15:31:54,328 [ search_worker_2][        search][    INFO] Searching s
tep 47, remaining 170
2016-08-02 15:31:55,128 [ search_worker_2][        models][    INFO] Upserted 0
pokemon, 30 pokestops, and 4 gyms
2016-08-02 15:31:55,157 [ search_worker_0][        search][    INFO] Searching s
tep 48, remaining 169
2016-08-02 15:31:55,920 [ search_worker_0][        models][    INFO] Upserted 0
pokemon, 30 pokestops, and 4 gyms
2016-08-02 15:31:56,288 [ search_worker_4][        search][    INFO] Searching s
tep 49, remaining 168
2016-08-02 15:31:57,155 [ search_worker_4][        models][    INFO] Upserted 1
pokemon, 32 pokestops, and 4 gyms
2016-08-02 15:31:58,664 [ search_worker_3][        search][    INFO] Searching s
tep 50, remaining 167
2016-08-02 15:31:59,364 [ search_worker_1][        search][    INFO] Searching s
tep 51, remaining 166
2016-08-02 15:31:59,520 [ search_worker_3][        models][    INFO] Upserted 0
pokemon, 32 pokestops, and 4 gyms
2016-08-02 15:32:00,140 [ search_worker_1][        models][    INFO] Upserted 0
pokemon, 48 pokestops, and 6 gyms
2016-08-02 15:32:00,309 [ search_worker_2][        search][    INFO] Searching s
tep 52, remaining 165
2016-08-02 15:32:00,954 [ search_worker_2][        models][    INFO] Upserted 0
pokemon, 48 pokestops, and 6 gyms
2016-08-02 15:32:01,052 [ search_worker_0][        search][    INFO] Searching s
tep 53, remaining 164
2016-08-02 15:32:01,829 [ search_worker_0][        models][    INFO] Upserted 0
pokemon, 45 pokestops, and 6 gyms
2016-08-02 15:32:02,269 [ search_worker_4][        search][    INFO] Searching s
tep 54, remaining 163
2016-08-02 15:32:03,039 [ search_worker_4][        models][    INFO] Upserted 0
pokemon, 45 pokestops, and 6 gyms
2016-08-02 15:32:04,644 [ search_worker_3][        search][    INFO] Searching s
tep 55, remaining 162
2016-08-02 15:32:05,177 [ search_worker_1][        search][    INFO] Searching s
tep 56, remaining 161
2016-08-02 15:32:05,457 [ search_worker_3][        models][    INFO] Upserted 0
pokemon, 45 pokestops, and 6 gyms
2016-08-02 15:32:05,986 [ search_worker_1][        models][    INFO] Upserted 0
pokemon, 40 pokestops, and 4 gyms
2016-08-02 15:32:06,013 [ search_worker_2][        search][    INFO] Searching s
tep 57, remaining 160

**Comment by wardude202** [comment]
@Silvikk  how many threads are you running though?

**Comment by Syndiction** [comment]
@wardude202 threads no longer exist in current dev build

**Comment by wardude202** [comment]
@Syndiction  so its going to be 1 circle scan every 5 seconds? (slow)?  by default?

**Comment by Silvikk** [comment]
I have 5 accs and there are 5 threads automatically.

**Comment by Syndiction** [comment]
@wardude202 each account is now utilized as a thread

**Comment by wardude202** [comment]
they should add something showing each account logging in

**Comment by wardude202** [comment]
@Silvikk i have 3 accounts in the config.ini file and its still slow as balls (1 circle every 5 seconds) 

Config
# Authentication settings
auth-service: ptc
username:  User
password:  Pass

auth-service: ptc
username: User
password: Pass

auth-service: google
username: User
password: Pass

**Comment by Syndiction** [comment]
From what i'm seeing you have to put them into the cmd, not the config file

**Comment by Silvikk** [comment]
@wardude202 I don't have a config file, but everything works...somehow :D Yaeh, put them into cmd!

2016-08-02 15:44:11,401 [   search_thread][        search][    INFO] Starting se
arch worker threads
2016-08-02 15:44:11,411 [   search_thread][        search][    INFO] New locatio
n caught, moving search grid
2016-08-02 15:44:11,414 [ search_worker_0][        search][    INFO] Searching s
tep 1, remaining 0
2016-08-02 15:44:11,420 [ search_worker_2][        search][    INFO] Searching s
tep 2, remaining 154
2016-08-02 15:44:11,420 [ search_worker_4][        search][    INFO] Searching s
tep 3, remaining 154
2016-08-02 15:44:11,421 [ search_worker_1][        search][    INFO] Searching s
tep 4, remaining 155
2016-08-02 15:44:11,421 [ search_worker_3][        search][    INFO] Searching s
tep 5, remaining 155
2016-08-02 15:44:11,437 [ search_worker_0][      auth_ptc][    INFO] PTC login f
or: ********
2016-08-02 15:44:11,437 [ search_worker_2][      auth_ptc][    INFO] PTC login f
or: ********
2016-08-02 15:44:11,437 [ search_worker_4][      auth_ptc][    INFO] PTC login f
or: ********
2016-08-02 15:44:11,451 [ search_worker_1][      auth_ptc][    INFO] PTC login f
or: ********
2016-08-02 15:44:11,460 [ search_worker_3][      auth_ptc][    INFO] PTC login f
or: ********
2016-08-02 15:44:14,601 [ search_worker_2][      auth_ptc][    INFO] PTC Login s
uccessful
2016-08-02 15:44:14,683 [ search_worker_1][      auth_ptc][    INFO] PTC Login s
uccessful
2016-08-02 15:44:15,561 [ search_worker_0][      auth_ptc][    INFO] PTC Login s
uccessful
2016-08-02 15:44:16,023 [ search_worker_3][      auth_ptc][    INFO] PTC Login s
uccessful

**Comment by wardude202** [comment]
@Silvikk  isnt this all running on Grunt??? (git bash)?

**Comment by Syndiction** [comment]
@wardude202 its still a CMD of sorts :P

**Comment by Silvikk** [comment]
@wardude202 
1. pip install -r requirements.txt
2. npm install
3. put a proper command into cmd

**Comment by wardude202** [comment]
@SilverC  so im doing this all in normal cmd not git bash

**Comment by Syndiction** [comment]
@wardude202  It doesnt really make a difference

**Comment by wardude202** [comment]
okay so i think its sorta working, but its still only scanning 1 aria every 5-6 seconds? logged in with at least 2 accounts from what i can see

**Comment by wardude202** [comment]
python runserver.py -a ptc -u -u -p -p -a google -u -p -l "-xxxyxxxx, yyyyyyyy" -st 7 

that should work right?

**Comment by Syndiction** [comment]
@wardude202 From what i understand yes

**Comment by Syndiction** [comment]
Once i get this build working i'm going to attempt with 7 accounts to see if it improves speed

**Comment by wardude202** [comment]
so yeah.... 1 account is the same speed as 3 for me.. NO idea why...

**Comment by wardude202** [comment]
i tested the config file by adding another fake account and the scan wouldn't even start, (its only using my last account in the config file)

**Comment by Syndiction** [comment]
@wardude202 in config they should be on the same line with commas to seperate them, thats what i have understood from the research i've done

**Comment by wardude202** [comment]
auth-service: ptc
username: USERA, USERB
password: PASSA, PASSB

auth-service: google
username: USERA@gmail.com, USERB
password: PASSA, PASSB

Like that?

**Comment by Syndiction** [comment]
@wardude202 yes, except from what i read they need to all have just one password if used in config

**Comment by wardude202** [comment]
@Scyntrus  just tried this, and it tries to log into the account with the user USERA, USERB instead of just USERA so that theory doesent work

**Comment by 1111121RicardoBrandao** [comment]
Yes is only using the last account because you are only setting one

you should do like this

auth-service: service
username: [user1, user2]
password: [pass1, pass2]

if the passwords are the same for both accounts you dont need to do password: [pass1, pass2] you can do password: pass

**Comment by wardude202** [comment]
How do i go abbout using a PTC and Google account? all in the config?

**Comment by 1111121RicardoBrandao** [comment]
Yes use config.ini i didnt test it with both logins but should be fine

**Comment by wardude202** [comment]
just got an error 
'error: username can't be set to a list '[USERA, USERB]" unless its action type is changed to 'append'

**Comment by Syndiction** [comment]
Do they have to be in the brackets?

**Comment by 1111121RicardoBrandao** [comment]
Are you using master or develop branch? The change to multiple accounts is not yet on master branch

**Comment by Syndiction** [comment]
dev

**Comment by wardude202** [comment]
Im using the dev branch

**Comment by wardude202** [comment]
@bengalaviz  still getting the same error as before :/

**Comment by wardude202** [comment]
FOR FUCK SAKES.. just reliased it closed master files NOT! dev FUCK SAKES.. hold on guys

**Comment by 1111121RicardoBrandao** [comment]
@wardude202 for what i see from your copy paste code, looks that you are using the master branch and not the develop branch

@Syndiction yes you need the brackets, its a list of accounts

**Comment by Syndiction** [comment]
if i can get my NPM install to work... i'll test it as soon as i can

**Comment by jhonatansm** [comment]
sorry, what is ptc account?
where i register?

**Comment by wardude202** [comment]
ALL FIXED and working thanks to the guys who helped! turns out my fucking command was cloning the normal master branch not the develop one, not sure what to change in the command for the dev one

**Comment by 1111121RicardoBrandao** [comment]
@jhonatansm Poke trainer club

**Comment by Syndiction** [comment]
@wardude202 is it going faster for you?

**Comment by wardude202** [comment]
@Scyntrus  MUCH faster

**Comment by Syndiction** [comment]
So if i can get it running with 7 accounts i should be golden :)

**Comment by wardude202** [comment]
@Scyntrus  pretty much, i have it running on my SERVER 24/7 and have shared it with a couple of mates

**Comment by Syndiction** [comment]
Cant wait

**Comment by Syndiction** [comment]
I finally got it all working, had to do a NPM cache clear because it kept erroring out, its so extremely fast now its awesome

**Comment by jhonatansm** [comment]
then I'm doing wrong, I set three google acc, I change a PTC?

**Comment by Syndiction** [comment]
google should work fine in the dev version if you follow what was said in @bengalaviz post

**Comment by Syndiction** [comment]
running 8 now, with 7 ptc and 1 google, it works very nicely, the first 4 circles are cleared in less than 10 seconds

**Comment by jhonatansm** [comment]
@Syndiction 

questions:
1. You can share your command to run?
2. you use the latest version 2.1.0?
3. or you use the dev version?

**Comment by Syndiction** [comment]
Dev is the only one with this feature,and its all in the config file, all i do for cmd is "python runserver.py"

**Comment by brianhrowe** [comment]
@Syndiction when I try it this way I get an error saying that -k --gmaps-key is required...

**Comment by Syndiction** [comment]
Config.ini

# Authentication settings
auth-service: [ptc, ptc, ptc, ptc, ptc, ptc, ptc, google]
username: [UserName1, UserName2, UserName3, UserName4, UserName5, UserName6, UserName7, Google1]
password: password

# Search settings
location: 
scan-delay: 5           # default 5
step-limit: 15            # default 12

# Misc
gmaps-key: (Required)          

# Webserver settings
#host:                  # address to listen on (default 127.0.0.1)
#port:                  # port to listen on (default 5000)

**Comment by brianhrowe** [comment]
@Syndiction My file looks like this...

# Authentication settings
auth-service: [ptc, ptc, ptc]
username: [ptc1, ptc2, ptc3]
password: password

# Search settings
location: "Location"
#no-gyms:               # disables gym scanning (default false)
#no-pokemon:            # disables pokemon scanning (default false)
#no-pokestops:          # disables pokestop scanning (default false)
#scan-delay:            # default 5
#step-limit:            # default 12

# Misc
gmaps-key: key
#webhook:               # webhook URL (including http://)

returns this when I run  `python runserver.py` ...

> usage: runserver.py [-h] [-a AUTH_SERVICE] [-u USERNAME] [-p PASSWORD]
                    [-l LOCATION] [-st STEP_LIMIT] [-sd SCAN_DELAY]
                    [-ld LOGIN_DELAY] [-dc] [-H HOST] [-P PORT] [-L LOCALE]
                    [-c] [-d] [-m] [-ns] [-os] [-nsc] [-fl] -k GMAPS_KEY [-C]
                    [-D DB] [-cd] [-np] [-ng] [-nk] [--db-type DB_TYPE]
                    [--db-name DB_NAME] [--db-user DB_USER]
                    [--db-pass DB_PASS] [--db-host DB_HOST]
                    [--db-max_connections DB_MAX_CONNECTIONS]
                    [-wh [WEBHOOKS [WEBHOOKS ...]]]
runserver.py: error: argument -k/--gmaps-key is required

**Comment by Syndiction** [comment]
Is your gmaps key set up correctly? like in the wiki?

**Comment by jhonatansm** [comment]
@Syndiction  brother help me with this

Command "python setup.py egg_info" failed with e
s5\appdata\local\temp\pip-build-dydh7t\pyproj\

when run setup easy

**Comment by brianhrowe** [comment]
@Syndiction yes it works when I use it in the command line

**Comment by Syndiction** [comment]
@jhonatansm you might need a full pip uninstall run "pip uninstall -r -y requirements.txt" then rerun "pip install -r requirements.txt" then do a "npm cache clean" then "npm install -g grunt" and finally "npm install" after everything runs you should be able to run everything regardless of errors

@brianhrowe Do what i just suggested for @jhonatansm and if that doesnt work keep the key in your cmd with -k it might be misreading your config file

**Comment by jhonatansm** [comment]
more errors
2016-08-02 11:06:00,621 [      MainThread]
"grunt build" before starting the server

**Comment by Syndiction** [comment]
@jhonatansm did you do every recode as i typed it?

**Comment by jhonatansm** [comment]
@Syndiction I did not do it

**Comment by jhonatansm** [comment]
@Syndiction  sorry last error

 File "runserver.py", line 15, in <modul
   from flask_cors import CORS
mportError: No module named flask_cors

when install  flask_cors with pip show error too

` File "runserver.py", line 15, in <modul
   from flask_cors import CORS
mportError: No module named flask_cors`

**Comment by Syndiction** [comment]
Did you do the full uninstall of the pip files? this shouldn't happen if you followed my steps

**Comment by jhonatansm** [comment]
I'm following in your steps

```
C:\PokemonGo-Map>pip uninstall -r requirements.txt
Cannot uninstall requirement Flask, not installed
----------------------
C:\PogoMap>pip uninstall -r requirements.txt
Cannot uninstall requirement six, not installed
```

**Comment by jhonatansm** [comment]
error is flask 

http://subefotos.com/ver/?3f4d99def7eaf36e063d2de65e3df25ao.png

**Comment by jhonatansm** [comment]
@Syndiction  works full all xD

regards

**Comment by WtfJoke** [comment]
> my fucking command was cloning the normal master branch not the develop one

`git checkout develop`
after cloning the master one....

or directly
`git clone -b develop https://github.com/AHAAAAAAA/PokemonGo-Map.git`
